core/sql/generator/GenPreCode.cpp

```diff
@@ -4385,12 +4377,14 @@ RelExpr * GenericUpdate::preCodeGen(Generator * generator,
 	  GenExit();
 	}
 
-      if ((getOperatorType() == REL_HBASE_DELETE) &&
+      if ((getOperatorType() != REL_HBASE_UPDATE) &&
           (mergeInsertRecExpr().entries() > 0) &&
           (CmpCommon::getDefault(COMP_BOOL_175) == DF_OFF))
 	{
+          // MERGE with INSERT is limited to HBase updates unless
+          // the CQD is on
 	  *CmpCommon::diags() << DgSqlCode(-3241)
-			      << DgString0(" MERGE delete not allowed with INSERT.");
+			      << DgString0(" This MERGE is not allowed with INSERT.");
```

Comment:
Actually, the message is correct as-is. It is preventing MERGE DELETE + INSERT on Trafodion tables as well as non-Trafodion tables.